bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -1,244 +1,58 @@
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
-import org.eclipse.smarthome.core.events.EventPublisher;
-import org.osgi.service.component.annotations.Activate;
+import org.eclipse.smarthome.io.transport.mqtt.internal.MqttBrokerConnectionServiceInstance;
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Modified;
-import org.osgi.service.component.annotations.Reference;
-import org.osgi.service.component.annotations.ReferenceCardinality;
-import org.osgi.service.component.annotations.ReferencePolicy;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * MQTT Service for creating new connections to MQTT brokers from the Smart Home configuration file and registering
- * message publishers and subscribers. This service is the main entry point for all bundles wanting to use the MQTT
- * transport.
+ * This service allows you to enumerate system-wide configured Mqtt broker connections. You do not need this service
+ * if you want to manage own/local Mqtt broker connections. If you add a broker connection, it will be
+ * available immediately for all MqttService users. A removed broker connection may still be in use by consuming
+ * services.
+ *
+ * Added/removed connections are not permanent. If the service is shutdown/restored it will
+ * contain the system-wide configured connections again (usually via text files).
+ *
+ * System broker connections are not configured via this service. See {@link MqttBrokerConnectionServiceInstance}
+ * instead.
  *
  * @author David Graeff - Added/Removed observer interface, Add/Remove/Enumerate broker connections.
  * @author Davy Vanherbergen
  * @author Markus Rathgeb - Synchronize access to broker connections
  */
-@Component(immediate = true, service = {
-        MqttService.class }, configurationPid = "org.eclipse.smarthome.mqtt", property = {
-                "service.pid=org.eclipse.smarthome.mqtt" })
+@Component(immediate = true, service = { MqttService.class }, configurationPid = {
+        "org.eclipse.smarthome.mqtt" }, property = "service.pid=org.eclipse.smarthome.mqtt")
 @NonNullByDefault
 public class MqttService {
-    public static class Config {
-        public final @Nullable String name;
-        public final @Nullable String url;
-        public final @Nullable String user;
-        public final @Nullable String pwd;
-        public final @Nullable String clientId;
-        public final @Nullable Integer keepAlive;
-        public final @Nullable Integer qos;
-        public final @Nullable Boolean retain;
-        public final @Nullable String lwt;
-        public final @Nullable String lwtTopic;
-        public final byte @Nullable [] lwtMessage;
-        public final @Nullable Integer lwtQos;
-        public final @Nullable Boolean lwtRetain;
```

Comment:
Please review the latest changes @maggu2810 did on the LWT configuration. It is now possible to either pass the encoded String with `topic:message:qos:retained` or provide all values in separate fields (see https://github.com/eclipse/smarthome/pull/5231).